content/blog/production-ready-object-detection-model-training-workflow-with-hpe-machine-learning-development-environment.md

```diff
@@ -0,0 +1,1099 @@
+```
+
+```python
+# unzip images and labels from /home/ubuntu/e2e_blogposts/ngc_blog
+!tar -xf train_images.tgz -C xview_dataset/
+```
+
+```python
+# unzip labels from /home/ubuntu/e2e_blogposts/ngc_blog directory 
+!tar -xf train_labels.tgz -C xview_dataset/
+```
+
+## 1. Convert TIF to RGB
+
+```python
+# Here loop through all the images and convert them to RGB, this is important for tiling the images and training with pytorch
+# will take about an hour to complete
+!python data_utils/tif_2_rgb.py --input_dir xview_dataset/train_images \
+  --out_dir xview_dataset/train_images_rgb/
+```
+
+## 2. How to convert labels to coco format
+
+Here we run a script to convert the dataset labels from .geojson format to COCO format. [Read more details about the COCO format at this link.](https://www.immersivelimit.com/tutorials/create-coco-annotations-from-scratch)
+
+The result will be two files (in COCO formal) generated `train.json` and `val.json`
+
+```python
+# make sure train_images_dir is pointing to the .tif images
+!python data_utils/convert_geojson_to_coco.py --train_images_dir xview_dataset/train_images/ \
+  --train_images_dir_rgb xview_dataset/train_images_rgb/ \
+  --train_geojson_path xview_dataset/xView_train.geojson \
+  --output_dir xview_dataset/ \
+  --train_split_rate 0.75 \
+  --category_id_remapping data_utils/category_id_mapping.json \
+  --xview_class_labels data_utils/xview_class_labels.txt
+```
+
+## 3. Slicing/Tiling the Dataset
+
+Here we are using the SAHI library to slice our large satellite images. Satellite images can be up to 50k^2 pixels in size, which wouldn't fit in GPU memory. We alleviate this problem by slicing the image. 
+
+```python
+!python data_utils/slice_coco.py --image_dir xview_dataset/train_images_rgb/ \
+  --train_dataset_json_path xview_dataset/train.json \
+  --val_dataset_json_path xview_dataset/val.json \
+  --slice_size 300 \
+  --overlap_ratio 0.2 \
+  --ignore_negative_samples True \
+  --min_area_ratio 0.1 \
+  --output_train_dir xview_dataset/train_images_rgb_no_neg/ \
+  --output_val_dir xview_dataset/val_images_rgb_no_neg/
+```
+
+## 4. Upload to s3 bucket to support distributed training
+
+We will now upload our exported data to a publically accessible S3 bucket. This will enable for a large scale distributed experiment to have access to the dataset without installing the dataset on device. 
+View [Determined Documentation](<* https://docs.determined.ai/latest/training/load-model-data.html#streaming-from-object-storage>) and [AWS instructions](<* https://codingsight.com/upload-files-to-aws-s3-with-the-aws-cli/>) to learn how to upload your dataset to an S3 bucket. Review the `S3Backend` class in `data.py`
```

Comment:
Hi Alex, Please remove the "<*" and ">" characters when you hotlink the "Determined Documentation" and "AWS instructions"